File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/ontology_graph_node.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class OntologyGraphNode(object):
    """
    Published ontology graph node.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new OntologyGraphNode object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param id:
            The value to assign to the id property of this OntologyGraphNode.
        :type id: str

        :param type:
            The value to assign to the type property of this OntologyGraphNode.
        :type type: str

        :param name:
            The value to assign to the name property of this OntologyGraphNode.
        :type name: str

        :param label:
            The value to assign to the label property of this OntologyGraphNode.
        :type label: str

        :param description:
            The value to assign to the description property of this OntologyGraphNode.
        :type description: str

        """
        self.swagger_types = {
            'id': 'str',
            'type': 'str',
            'name': 'str',
            'label': 'str',
            'description': 'str'
        }

        self.attribute_map = {
            'id': 'id',
            'type': 'type',
            'name': 'name',
            'label': 'label',
            'description': 'description'
        }

        self._id = None
        self._type = None
        self._name = None
        self._label = None
        self._description = None

    @property
    def id(self):
        """
        **[Required]** Gets the id of this OntologyGraphNode.

        :return: The id of this OntologyGraphNode.
        :rtype: str
        """
        return self._id

    @id.setter
    def id(self, id):
        """
        Sets the id of this OntologyGraphNode.

        :param id: The id of this OntologyGraphNode.
        :type: str
        """
        self._id = id

    @property
    def type(self):
        """
        **[Required]** Gets the type of this OntologyGraphNode.
        CLASS, PROPERTY, SHAPE, CONSTRAINT, or RELATIONSHIP.


        :return: The type of this OntologyGraphNode.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this OntologyGraphNode.
        CLASS, PROPERTY, SHAPE, CONSTRAINT, or RELATIONSHIP.


        :param type: The type of this OntologyGraphNode.
        :type: str
        """
        self._type = type

    @property
    def name(self):
        """
        Gets the name of this OntologyGraphNode.

        :return: The name of this OntologyGraphNode.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this OntologyGraphNode.

        :param name: The name of this OntologyGraphNode.
        :type: str
        """
        self._name = name

    @property
    def label(self):
        """
        Gets the label of this OntologyGraphNode.

        :return: The label of this OntologyGraphNode.
        :rtype: str
        """
        return self._label

    @label.setter
    def label(self, label):
        """
        Sets the label of this OntologyGraphNode.

        :param label: The label of this OntologyGraphNode.
        :type: str
        """
        self._label = label

    @property
    def description(self):
        """
        Gets the description of this OntologyGraphNode.

        :return: The description of this OntologyGraphNode.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this OntologyGraphNode.

        :param description: The description of this OntologyGraphNode.
        :type: str
        """
        self._description = description

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/OntologyGraphNode.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Published ontology graph node.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=OntologyGraphNode.Builder.class)

public final class OntologyGraphNode  {
    @Deprecated
    @java.beans.ConstructorProperties({"id", "type", "name", "label", "description"})
    public OntologyGraphNode(String id, String type, String name, String label, String description) {
        super();
        this.id = id;
        this.type = type;
        this.name = name;
        this.label = label;
        this.description = description;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            
@com.fasterxml.jackson.annotation.JsonProperty("id")
private String id;



public Builder id(String id) {
    this.id = id;
    return this;
}
            /**
     * CLASS, PROPERTY, SHAPE, CONSTRAINT, or RELATIONSHIP.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private String type;

        /**
         * CLASS, PROPERTY, SHAPE, CONSTRAINT, or RELATIONSHIP.
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(String type) {
    this.type = type;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;



public Builder name(String name) {
    this.name = name;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("label")
private String label;



public Builder label(String label) {
    this.label = label;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;



public Builder description(String description) {
    this.description = description;
    return this;
}


        public OntologyGraphNode build() {
            OntologyGraphNode model = new OntologyGraphNode(this.id
                , this.type
                , this.name
                , this.label
                , this.description);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(OntologyGraphNode model) {
                this.id(model.getId());
    this.type(model.getType());
    this.name(model.getName());
    this.label(model.getLabel());
    this.description(model.getDescription());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


    
    @com.fasterxml.jackson.annotation.JsonProperty("id")
    private final String id;

    
    public String getId() {
        return id;
    }


        /**
     * CLASS, PROPERTY, SHAPE, CONSTRAINT, or RELATIONSHIP.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final String type;

        /**
     * CLASS, PROPERTY, SHAPE, CONSTRAINT, or RELATIONSHIP.
     * @return the value
     **/
    
    public String getType() {
        return type;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

    
    public String getName() {
        return name;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("label")
    private final String label;

    
    public String getLabel() {
        return label;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

    
    public String getDescription() {
        return description;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("OntologyGraphNode(");
        sb.append("id=").append(String.valueOf(this.id));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", label=").append(String.valueOf(this.label));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof OntologyGraphNode)) {
            return false;
        }

        OntologyGraphNode other = (OntologyGraphNode) o;
        return java.util.Objects.equals(this.id, other.id) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.label, other.label) &&
            java.util.Objects.equals(this.description, other.description);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.id == null ? 43 : this.id.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.label == null ? 43 : this.label.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        return result;
    }


}
